beetsplug/lastgenre/__init__.py

```diff
@@ -314,8 +314,19 @@ def _get_genre(self, obj):
         """
 
         # Shortcut to existing genre if not forcing.
-        if not self.config["force"] and self._is_allowed(obj.genre):
-            return obj.genre, "keep"
+        if not self.config["force"]:
```

Comment:
Sorry to be a little slow here, but could you explain a little bit more about what the intended behavior of the new code is when `force` is off? It kinda looks like it modifies genre information that is already in the database, which may not be intended…

That is, say my current genre is `foo, bar`. When I run the plugin with `force` off, I'd kinda expect it to stay `foo, bar`. But if I'm understanding correctly, it would in fact filter the list of genres to keep only the "allowed" genres. So it might change the genre to `foo` to reflect my current allowlist.

While that seems a little weird, I actually am having trouble figuring out anything else that would be less weird… keeping all the genres if at least one is allowed seems weird too? So maybe this is the right thing to do, but it's weird enough that maybe we want to document it carefully.

Reply:
Thanks for thoroughly thinking it through, this new behaviour is indeed still a little weird. But let's go back to what it did before: Since a string `foo, bar` did not match the whitelist, it was kicked out entirely. If it would have been just `foo` or just `bar` it would have been kept. That was even more weird.

So now it at least only kicks out non-whitelist genres of the multi-genre string. It kind of does a cleanup.

But I totally agree that is still not exactly what one would expect from a `force: no` setting....so yeah documenting carefully seems essential at least...but let me think harder about it for a day or two :-)

Reply:
Sounds good!

Reply:
Two days can even to be two weeks ;-) but I have an idea how we might make `force: no` more predictable. It requires a new config setting as well as a whole new branch of behaviour 

force: yes
overwrite all, only fresh last.fm genres remain

force: no
keep any string in present genre tag, only write last.fm genres when empty

force: yes
keep_allowed: yes
keep present genres when whitelisted and add new last.fm genres (this is a new branch of behaviour and needs to be coded, There is open feature requests for it. fixme: I 'll check!

force: no
keep_allowed: yes
cleanup only - keep present genres when whitelisted but don't add new last.fm genres

Reply:
Let's discuss this further in a new "main" thread -> https://github.com/beetbox/beets/pull/4982#issuecomment-1815974394